com/williamfiset/algorithms/dp/VirtualMachinesScheduler.java

```diff
@@ -0,0 +1,177 @@
+package com.williamfiset.algorithms.dp;
+
+import java.io.FileReader;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Scanner;
+
+/**
+ * Cloud computing companies, like Amazon, have a lot of virtual machines in their disposal, which
+ * they offer to users for a price for the total time of the VM usage. There are quite a few
+ * different types of VMs with different CPU and Disk speeds as well as storage sizes, which all
+ * have different costs. For example VM-A has faster CPUs than VM-B, but VM-B has faster storage
+ * read/write. Then, there is VM-C which is faster than both VM-A and VM-B in both CPU and Disk
+ * speeds but is considerably more expensive. The problem that arises is how to assign the processes
+ * to the VMs efficiently and we achieve that with dynamic programming.
```

Comment:
What problem does this generalize as? I can't imagine the only application is for scheduling VMs

Reply:
It's a scheduling algorithm. With modifications it could be used in other scheduling problems too, like assigning jobs to employees etc.